### ISSUE TITLE
Phone lib addon - parentheses

### ISSUE THREAD

**Issue by zslabs** [nosir/cleave.js#344]
I wasn't able to find any issues around this, but wanted to know if there was ever any talk about adding more formatting options around a specific country code's formatting. For example, a US number looks like this currently: `555 555 5555` but it'd be great if there was an option to surround that text like so `(555) 555-5555`.

**Comment by LentaTillux** [comment]
my version of the "crutch."
but delimiters does not support " " (there is a space) and 2 characters in a row ") " or ")-"
when 2 characters, then deletion <- occurs before the place ") " and further blocked, - you must manually rearrange the cursor

```
var phoneMask = {
            prefix: '+7 (',
            delimiters: [')', '-', '-'],
            blocks: [7, 3, 2, 2],
            uppercase: true
        };

**Comment by LentaTillux** [comment]
if  could add 2 characters and a space character, you would get support for large data formats

**Comment by Timethor** [comment]
My company is considering not using this library for phone masking because most of our customers require parentheses for their style guides... would love to not have to integrate with another lib for this one use case.

I'm willing to spend some time on a PR with some direction from maintainers if there is interest.

**Comment by nosir** [comment]
@zslabs Does this work for you?
https://jsfiddle.net/qnmds8dn/2/
```
new Cleave("input", {
	numericOnly: true,
	blocks: [0, 3, 3, 4],
  	delimiters: ['(', ') ', '-']
});
```

**Comment by zslabs** [comment]
@nosir That actually works pretty well! I noticed that if I type a full number and attempt to backspace; it gets "stuck" before the  `) `. Happening to you at all?

**Comment by stephenjwatkins** [comment]
Until #348 is addressed, I'm adding an extra block and delimiter for the space. It's not the ideal aesthetic (since the space comes after the next number), but I found it more important to have the ability to backspace.

```js
new Cleave(this.el, {
  numericOnly: true,
  blocks: [0, 3, 0, 3, 4],
  delimiters: ["(", ")", " ", "-"],
});
```

**Comment by nosir** [comment]
Ah cool, I shall add this to demo

**Comment by nosir** [comment]
Added @stephenjwatkins 's solution to the demo: https://jsfiddle.net/nosir/aLnhdf3z/

**Comment by Schw2iizer** [comment]
@nosir Any reason why this wouldn't work in a React app? My code looks like this but no blocks and delimiters updates. 

<Cleave placeholder={this.props.placeholder}
              className={this.props.className}
              options={{
                phone: true,
                phoneRegionCode: 'US',
                blocks:[0, 3, 0, 3, 4],
                delimiters: ['(', ')', ' ', '-']
              }}
              />

**Comment by nosir** [comment]
@Schw2iizer To use custom `blocks` or `phone: true`, you can only choose one option, they don't work together, by using `phone: true`, it resolves blocks automatically for you, and in that case, you can only define single delimiter.
TLDR;
either:
```
options={{
blocks:[0, 3, 0, 3, 4],
delimiters: ['(', ')', ' ', '-']
}}
```

or 

```
options={{
phone: true,
phoneRegionCode: 'US',
delimiter: '-'
}}
```

**Comment by Schw2iizer** [comment]
@nosir That's great. Thank you for that info.

**Comment by EricRoos** [comment]
Confirmed that suggestion by @nosir works as of today. Thank you.
